commit cf7c9078a5acdbb435498ace92cd81009637a971
Author: Ulrich Drepper <drepper@redhat.com>
Date:   Fri Nov 19 09:35:39 2010 -0500

    Define MAP_HUGETLB.

	* sysdeps/unix/sysv/linux/i386/bits/mman.h: Define MAP_HUGETLB.
	* sysdeps/unix/sysv/linux/x86_64/bits/mman.h: Likewise.

commit dd62fda6cc83b3afa150a4be7d273ab118bf9825
Author: Anton Blanchard <anton@samba.org>
Date:   Tue Mar 27 18:37:16 2012 -0700

    Define MAP_STACK and MAP_HUGETLB on powerpc, s390, sh and sparc.
    
        * sysdeps/unix/sysv/linux/powerpc/bits/mman.h: Define MAP_STACK and
        MAP_HUGETLB.
        * sysdeps/unix/sysv/linux/s390/bits/mman.h: Likewise.
        * sysdeps/unix/sysv/linux/sh/bits/mman.h: Likewise.
        * sysdeps/unix/sysv/linux/sparc/bits/mman.h: Likewise.

commit 7e4fc5890e07549e5ad98b4244e3e0ed5c709e6a
Author: Mike Frysinger <vapier@gentoo.org>
Date:   Mon Apr 16 22:44:23 2012 -0400

    ia64: add MAP_{STACK,HUGETLB} to bits/mman.h

    Just adding new bit flags that newer kernels define.

    Signed-off-by: Mike Frysinger <vapier@gentoo.org>

	* sysdeps/unix/sysv/linux/ia64/bits/mman.h (MAP_STACK): Define.
	(MAP_HUGETLB): Likewise.

Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/powerpc/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/powerpc/bits/mman.h	2013-07-23 04:01:42.440052942 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/powerpc/bits/mman.h	2013-07-23 22:26:42.317789260 -0300
@@ -65,6 +65,8 @@
 # define MAP_NORESERVE	0x00040		/* Don't check for reservations.  */
 # define MAP_POPULATE	0x08000		/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	0x10000		/* Do not block on IO.  */
+# define MAP_STACK	0x20000		/* Allocation is for a stack.  */
+# define MAP_HUGETLB	0x40000		/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/s390/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/s390/bits/mman.h	2013-07-23 04:01:42.447052393 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/s390/bits/mman.h	2013-07-23 22:26:42.335787833 -0300
@@ -64,6 +64,8 @@
 # define MAP_NORESERVE	0x04000		/* Don't check for reservations.  */
 # define MAP_POPULATE	0x08000		/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	0x10000		/* Do not block on IO.  */
+# define MAP_STACK	0x20000		/* Allocation is for a stack.  */
+# define MAP_HUGETLB	0x40000		/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/sh/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/sh/bits/mman.h	2010-05-04 08:27:23.000000000 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/sh/bits/mman.h	2013-07-23 22:26:42.336787753 -0300
@@ -64,6 +64,8 @@
 # define MAP_NORESERVE	0x4000		/* Don't check for reservations.  */
 # define MAP_POPULATE	0x8000		/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	0x10000		/* Do not block on IO.  */
+# define MAP_STACK	0x20000		/* Allocation is for a stack.  */
+# define MAP_HUGETLB	0x40000		/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/sparc/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/sparc/bits/mman.h	2010-05-04 08:27:23.000000000 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/sparc/bits/mman.h	2013-07-23 22:26:42.337787674 -0300
@@ -66,6 +66,8 @@
 # define _MAP_NEW	0x80000000	/* Binary compatibility with SunOS.  */
 # define MAP_POPULATE	0x8000		/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	0x10000		/* Do not block on IO.  */
+# define MAP_STACK	0x20000		/* Allocation is for a stack.  */
+# define MAP_HUGETLB	0x40000		/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/i386/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/i386/bits/mman.h	2013-07-23 04:01:42.406055607 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/i386/bits/mman.h	2013-07-23 22:32:37.407630135 -0300
@@ -64,6 +64,7 @@
 # define MAP_POPULATE	0x08000		/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	0x10000		/* Do not block on IO.  */
 # define MAP_STACK	0x20000		/* Allocation is for a stack.  */
+# define MAP_HUGETLB	0x40000		/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/x86_64/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/x86_64/bits/mman.h	2013-07-23 04:01:42.493048788 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/x86_64/bits/mman.h	2013-07-23 22:32:37.433628073 -0300
@@ -65,6 +65,7 @@
 # define MAP_POPULATE	0x08000		/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	0x10000		/* Do not block on IO.  */
 # define MAP_STACK	0x20000		/* Allocation is for a stack.  */
+# define MAP_HUGETLB	0x40000		/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
Index: glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/ia64/bits/mman.h
===================================================================
--- glibc-2.12-2-gc4ccff1.orig/sysdeps/unix/sysv/linux/ia64/bits/mman.h	2010-05-04 08:27:23.000000000 -0300
+++ glibc-2.12-2-gc4ccff1/sysdeps/unix/sysv/linux/ia64/bits/mman.h	2013-07-23 22:37:28.643530951 -0300
@@ -65,6 +65,8 @@
 # define MAP_NORESERVE	  0x04000	/* Don't check for reservations.  */
 # define MAP_POPULATE	  0x08000	/* Populate (prefault) pagetables.  */
 # define MAP_NONBLOCK	  0x10000	/* Do not block on IO.  */
+# define MAP_STACK	  0x20000	/* Allocation is for a stack.  */
+# define MAP_HUGETLB	  0x40000	/* Create huge page mapping.  */
 #endif
 
 /* Flags to `msync'.  */
